Extension/src/LanguageServer/clientCollection.ts

```diff
@@ -198,13 +262,21 @@ export class ClientCollection {
             if (client) {
                 return client;
             }
-            const newClient: cpptools.Client = cpptools.createClient(this, folder);
-            this.languageClients.set(key, newClient);
-            getCustomConfigProviders().forEach(provider => newClient.onRegisterCustomConfigurationProvider(provider));
-            const defaultClient: cpptools.DefaultClient = <cpptools.DefaultClient>newClient;
-            defaultClient.sendAllSettings();
-            return newClient;
+            return this.createClient(folder);
+        }
+    }
+
+    public createClient(folder?: vscode.WorkspaceFolder, deactivated?: boolean): cpptools.Client {
+        const newClient: cpptools.Client = cpptools.createClient(this, folder);
+        if (deactivated) {
+            newClient.deactivate(); // e.g. prevent the current config from switching.
```

Comment:
I am still a little unclear on this bit here.  Should ALL clients be created in a deactivated state, and only activated in the context of processing an `activeDocumentChange` (and when we need to replace `this.activeClient` and there is a `this.activeDocument` present) ?

Reply:
I believe the effect of having the client become active is that the config for that workspace folder will be shown in the status bar -- so you just have to decide if that's what you want to happen in your context. Usually it would make sense for all clients except for some "active" one to be deactivated...but if you don't detect an active one you can use the 1st workspace folder.